libcodechecker/server/api/report_server.py

```diff
@@ -1996,38 +1988,38 @@ def massStoreRun(self, name, tag, version, b64zip, force):
                 for file_name in files:
                     file_ids[file_name] = file_path_to_id[file_name]
 
-                report_path_and_events = {}
-
                 # Store report.
                 for report in reports:
                     LOG.debug("Storing check results to the database.")
 
-                    checker_name = report.main['check_name']
-                    context = generic_package_context.get_context()
-                    severity_name = context.severity_map.get(checker_name,
-                                                             'UNSPECIFIED')
-                    severity = \
-                        Severity._NAMES_TO_VALUES[severity_name]
-
                     bug_paths, bug_events = \
                         store_handler.collect_paths_events(report, file_ids,
                                                            files)
 
                     LOG.debug("Storing report")
                     bug_id = report.main[
                         'issue_hash_content_of_line_in_context']
+                    if bug_id in hash_map_reports:
+                        old_report = hash_map_reports[bug_id][0]
+                        old_status = old_report.detection_status
+                        detection_status = 'reopened' \
+                            if old_status == 'resolved' else 'unresolved'
+                        run_history_time = old_report.detected_at
```

Comment:
Storage 1:
```cpp
int main() {
  sizeof(42);
}
```

Storage 2:
```cpp
int main() {
  sizeof(42);
  sizeof(43);
}
```

Storage 3:
```cpp
int main() {
  sizeof(42);
  sizeof(43);
  sizeof(44);
}
```
All reports have the same detection date.

:exclamation: The detection date of the first report shouldn't be changed on the third storage.